v1/linkml-schemas/name.yaml

```diff
@@ -17,34 +17,38 @@ default_range: string
 
 classes:
   NameClass:
+    description: Name object that represnts a user's claimed identity.  This will have a list of name parts.  Usually this will contain one or more [GivenNameClass](../GivenNameClass) name parts and one [FamilyNameClass](../FamilyNameClass) name part for a specific period of validity.
     mixins:
       - ValidityClass
-    attributes:
-      nameParts:
-        range: NamePartClass
-        multivalued: true
-        inlined_as_list: true
+    slots:
+      - nameParts
+      
   NamePartClass:
+    description: Abstract class to define a name part value and name part type for an optional validity period.
     abstract: false  # has to be false otherwise JSON-Schema doesn't include it, and validation fails :(
     mixins:
       - ValidityClass
     slots:
       - value
-    attributes:
-      type:
-        range: NamePartType
-        multivalued: false
+      - type
+
   GivenNameClass:
+    description: Name part object that represents a Given Name.  There may be multiple [NamePart](../NamePartClass) objects with a [NamePartType](../NamePartType) of `GivenName` within the representation of a [Name](../NameClass) for a specific period of validity.  These will appear in the order in which the user specified or the order in which they were read from official documentation.
```

Comment:
Technically this may contain space characters too. The example is people called "Anne Marie" where sometimes you get the compound name but without the hyphen. Likely not worth calling out, dunno why I mentioned it really. Not quite sure the level of detail to pitch this documentation!

Reply:
Currently can only happen through DCMAW, but won't once they implement the splitting code, but I get the point.  If you currently type 2 names in first name and 2 names in middle name in Passport CRI you end up with:

`      "name": [
        {
          "nameParts": [
            {
              "type": "GivenName",
              "value": "william"
            },
            {
              "type": "GivenName",
              "value": "james"
            },
            {
              "type": "GivenName",
              "value": "jermaine"
            },
            {
              "type": "GivenName",
              "value": "marcus"
            },
            {
              "type": "FamilyName",
              "value": "Test test test"
            }
          ]
        }
      ]
`

I think if we try to mention it, we could confuse matters more.  Maybe this is where we need our tech writer to help us out :-)

Reply:
Hah, yep. Most of these cases can only really arise once we give users the magical hypothetical "tell us your name and exactly how you would like us to represent it" user interface... and at least we now have a name for that (the "claimed identity credential issuer", although Kerr doesn't like that name so it may not last :-D )